lib/user-interface/react-app/src/components/chatbot/multi-chat.tsx

```diff
@@ -0,0 +1,292 @@
+  const [workspaces, setWorkspaces] = useState<WorkspaceItem[]>([]);
+  const [enableAddModels, setEnableAddModels] = useState(true);
+  const [llmToConfig, setLlmToConfig] = useState<ChatSession | undefined>(
+    undefined
+  );
+  const [showMetadata, setShowMetadata] = useState(false);
+
+  const { sendJsonMessage, readyState } = useWebSocket(socketUrl, {
+    share: true,
+    shouldReconnect: () => true,
+    onMessage: (payload) => {
+      // Check the session id for the response and update the corresponding session
+      const response: ChatBotMessageResponse = JSON.parse(payload.data);
+      const sessionId = response.data.sessionId;
+      const session = chatSessions.filter((c) => c.id === sessionId)[0];
+      if (session !== undefined) {
+        updateChatSessions(session, response);
+        setChatSessions([...chatSessions]);
+      }
+    },
+  });
+
+  useEffect(() => {
+    if (!appContext) return;
+    
+    setChatSessions([createNewSession(), createNewSession()]); // reset all chats
+    setEnableAddModels(true);
+    (async () => {
+      const apiClient = new ApiClient(appContext);
+      const [session, modelsResult, workspacesResult] = await Promise.all([
+        Auth.currentSession(),
+        apiClient.models.getModels(),
+        appContext?.config.rag_enabled
+          ? apiClient.workspaces.getWorkspaces()
+          : Promise.resolve<ApiResult<WorkspaceItem[]>>({ ok: true, data: [] }),
+      ]);
+
+      const jwtToken = session.getAccessToken().getJwtToken();
+
+      if (jwtToken) {
+        setSocketUrl(
+          `${appContext.config.websocket_endpoint}?token=${jwtToken}`
+        );
+      }
+
+      const models = ResultValue.ok(modelsResult)
+        ? modelsResult.data.filter(
+            (m) => m.inputModalities.includes(ChabotInputModality.Text) && m.outputModalities.includes(ChabotOutputModality.Text))
+        : [];
+      setModels(models);
+      const workspaces = ResultValue.ok(workspacesResult)
+        ? workspacesResult.data
+        : [];
+      setWorkspaces(workspaces);
+    })();
+  }, [appContext]);
+
+  const handleSendMessage = (message: string): void => {
+    // send message to each chat session
+    setEnableAddModels(false);
+    chatSessions.forEach((chatSession) => {
+      if (chatSession.running) return;
+      if (readyState !== ReadyState.OPEN) return;
+      ChatScrollState.userHasScrolled = false; // check this
+
+      const { name, provider } = OptionsHelper.parseValue(
+        chatSession.model?.value
+      );
+
+      const value = message.trim();
+      const request: ChatBotRunRequest = {
+        action: ChatBotAction.Run,
+        modelInterface: chatSession.modelMetadata!.interface,
+        data: {
+          modelName: name,
+          provider: provider,
+          sessionId: chatSession.id,
+          files: [],
+          workspaceId: undefined, // state.selectedWorkspace?.value, // need to move this to the chat
```

Comment:
ditto (for comment)

Reply:
I need to fix this, otherwise RAG does not work